std/assembly/date.ts

```diff
@@ -61,37 +62,60 @@ export class Date {
       let timeString: string;
       dateString = dateTimeString.substring(0, posT);
       timeString = dateTimeString.substring(posT + 1);
-      // parse the HH-MM-SS component
+      
+      // might end with an offset ("Z", "+05:30", "-08:00", etc.)
+      for (let i = timeString.length - 1; i >= 0; i--) {
+        let c = timeString.charCodeAt(i);
+        if (c == 90) { // Z
+          timeString = timeString.substring(0, i);
+          break;
+        } else if (c == 43 || c == 45) { // + or -
+          if (i == timeString.length - 1) {
+            throw new RangeError(E_INVALIDDATE);
+          }
+          let offsetParts = timeString.substring(i+1).split(":");
```

Comment:
```suggestion
          let offsetParts = timeString.substring(i + 1).split(":");
```

Reply:
Also, could you avoid `split`? And use `indexOf` and two `substring` instead. It will prevent the creation of a temporary array